app/src/main/java/chat/rocket/android/activity/ServerConfigActivity.java

```diff
@@ -77,14 +83,27 @@ protected void onPause() {
   }
 
   private void onRenderServerConfigError(ServerConfig config) {
-    if (config != null) {
+    if (config.getState() == ServerConfig.STATE_CONNECTION_ERROR) {
```

Comment:
The loop is mainly caused by calling `registerForPush` in sessionObserver's callback.
It would be fixed by moving `registerForPush` out of the Activity as mentioned in https://github.com/RocketChat/Rocket.Chat.Android/pull/124#discussion_r94358779